backend/api/v1/instance_service.go

```diff
@@ -1155,25 +1155,35 @@ func convertToStoreDataSourceExternalSecret(externalSecret *v1pb.DataSourceExter
 	if err := convertV1PbToStorePb(externalSecret, secret); err != nil {
 		return nil, status.Errorf(codes.Internal, "failed to convert external secret with error: %v", err.Error())
 	}
-	if secret.SecretName == "" || secret.PasswordKeyName == "" {
-		return nil, status.Errorf(codes.InvalidArgument, "missing secret name or key name")
-	}
-	if secret.SecretType == storepb.DataSourceExternalSecret_VAULT_KV_V2 {
+	switch secret.SecretType {
+	case storepb.DataSourceExternalSecret_VAULT_KV_V2:
 		if secret.Url == "" {
 			return nil, status.Errorf(codes.InvalidArgument, "missing Vault URL")
 		}
 		if secret.EngineName == "" {
 			return nil, status.Errorf(codes.InvalidArgument, "missing Vault engine name")
 		}
+		fallthrough
```

Comment:
Do we need a fallthrough?

Reply:
Need fallthrough to check the secret name and key name

Reply:
Not a big fan of fallthrough (IMO, mistake in Golang design) because it's easy to fall through a fallthrough.

I think we can do
```
if secret.SecretType == storepb.DataSourceExternalSecret_VAULT_KV_V2 {
		if secret.Url == "" {
			return nil, status.Errorf(codes.InvalidArgument, "missing Vault URL")
		}
		if secret.EngineName == "" {
			return nil, status.Errorf(codes.InvalidArgument, "missing Vault engine name")
		}
}
if secret.SecretType == storepb.DataSourceExternalSecret_VAULT_KV_V2 || secret.SecretType == storepb.DataSourceExternalSecret_AWS_SECRETS_MANAGER {
  // the fallthrough code.
}
```